plugins/catalog-backend/src/database/DefaultProcessingDatabase.ts

```diff
@@ -515,7 +518,40 @@ export class DefaultProcessingDatabase implements ProcessingDatabase {
       throw new NotFoundError(`Failed to schedule ${entityRef} for refresh`);
     }
   }
+  async refreshByRefreshKey(
+    txOpaque: Transaction,
+    options: RefreshByKeyOptions,
+  ) {
+    const tx = txOpaque as Knex.Transaction;
+    const { key } = options;
 
+    const rows = await tx<DbRefreshKeysRow>('refresh_keys')
+      .where({ key })
+      .select({
+        entity_ref: 'refresh_keys.entity_ref',
+      });
+
+    await Promise.all(rows.map(r => this.refresh(tx, r.entity_ref)));
+  }
+  async addRefreshKeys(
+    txOpaque: Transaction,
+    options: RefreshKeyOptions,
+  ): Promise<void> {
+    const tx = txOpaque as Knex.Transaction;
+    const { keys } = options;
+
+    await Promise.all(
+      keys.map(k => {
+        return tx<DbRefreshKeysRow>('refresh_keys')
+          .insert({
+            entity_ref: stringifyEntityRef(k.entity),
+            key: k.key,
+          })
+          .onConflict(['entity_ref', 'key'])
+          .ignore();
```

Comment:
I wasn't able to perform this with tx.batchInsert when I want to use an onConflict. Would it be better to check first if they exist and only do an insert if don't?